### ISSUE TITLE
Scheduled series recording skipping over episodes when Season/Episode data unavailable in Guide

### ISSUE THREAD

**Issue by jestergs** [jellyfin/jellyfin#5856]
**Describe the bug**
When scheduling a series recording in Jellyfin, there is erratic behavior where individual episodes in the series will (inaccurately) be left off of the recording schedule. I've noticed this only for series where there is no Season or Episode information in the EPG Guide (i.e. local news). Even though I have the options to record every episode, and not to check against episodes already in the library, clicking around in the guide and checking the scheduled recordings will somehow randomly change the episodes that are scheduled to be recorded. I am using XML guide data and have observed this behavior without refreshing the guide data. Screenshots below.

**System (please complete the following information):**
 - OS: Windows 10
 - Virtualization: N/A
 - Clients: N/A
 - Browser: Chrome 89, Edge 89
 - Jellyfin Version: 10.7.2
 - Playback: N/A 
 - Installed Plugins: none
 - Reverse Proxy: none
 - Base URL: none
 - Networking: Host
 - Storage: local

**To Reproduce**

1. Open Guide

2. Click on a episode that is part of a series

3. Schedule recording of series (see screenshot 1, notice both series and episode icons are red)
![Scheduled Recordings 1](https://user-images.githubusercontent.com/82554988/115155432-f51a4e80-a04d-11eb-98d7-2741991405c2.png)

4. Click on series settings, configure to record all episodes and deselect option to compare against episodes already in library (see screenshot 2, notice many episodes are detected but only the first is scheduled to record)
![Scheduled Recordings 2](https://user-images.githubusercontent.com/82554988/115155429-f481b800-a04d-11eb-9b84-53af12525d92.png)

5. Go to back to guide and select same episode that was selected in Step 2 (see screenshot 3, notice episode is no longer scheduled to record)
![Scheduled Recordings 3](https://user-images.githubusercontent.com/82554988/115155431-f51a4e80-a04d-11eb-975b-89ed56740cff.png)

6. Click on series settings (see screenshot 4, notice original episode is no longer scheduled to record)
![Scheduled Recordings 4](https://user-images.githubusercontent.com/82554988/115155430-f481b800-a04d-11eb-884a-b4049833e3f4.png)

**Expected behavior**
Expect all episodes to be scheduled to record (in the screenshots, all nightly episodes are "new", so they all should have been scheduled to begin with.
Do not expect individual episodes to be un-scheduled simply by clicking back and forth with the guide and scheduled recordings, without making any changes

**Screenshots**
Screenshots attached here

**Additional context**
I am using an OTA antenna connected to an HDHomerun 4k
Guide data is imported from XML file generated by zap2xml

**Comment by azeroth12** [comment]
This still seems to be an issue.

**Comment by dRuEFFECT** [comment]
still an issue
![image](https://user-images.githubusercontent.com/7904175/158491797-6ccda1cf-c91c-4504-88df-64055722449a.png)

**Comment by melendyk** [comment]
Yeah, still a problem for me too.  I was really hoping to see it fixed in 10.8.  It's strange for sure, because it's not all series that have the issue... I mostly notice (if not always) only with say daily news programs.  Another pattern I noticed is once the series was scheduled, the next day after a guide refresh, jellyfin always flagged the the furthest out episode to record, creating a never ending chase the last episode to record with each passing day and refresh of the guide, but rarely every actually recording any episode.  My next observation lead me down the path of the below work around.  I had observed, if I forced a purge of all guide data by using an empty xmltv file, then doing another guide refresh using a populated xmltv file, I'd get the "next" episode to flag as being recorded (butt none of the following days would ever work again due to back to what I call "chasing the latest episode in the guide).

I do have a work around that looks promising.  I've just implemented it.

This is how I have it working.
For the guide source, I use xmltv to load the schedule (not schedules direct)
1) A cron job runs at 2:58 AM that creates an empty xmltv.xml file.
2) A scheduled task in jellyfin at 3:00 AM for "Refresh guide" - This will completely wipe out the schedule loaded in jellyfin and no guide or channels will be visible until step 4.
3) Another cron job runs at 3:07 AM to pull the new guide info creating the xmltv.xml file with all the show data.
4) Finally, another scheduled task in jellyfin at 3:15 AM to "Refresh guide" again. - This will load the guide again back into jellyfin.
Note, this does leave a 15 minute window of no guide data or channels being visible, that is why I do it a 3AM.

When step 4 loads the guide, for the "series" that have had problem, it'll at least set the next one to record (shown in image below).  The following ones will still show as grey, not recording, however, then next run of 1-4 will catch the next one.  This is certainly not ideal, but seems to be a work around.  I would do it all in script and skip 2 and 4, but I can't seem to find an API call to "refresh guide."  If there was such an API call, my script would incorporate all steps in little package.

![Screenshot from 2022-06-14 09-36-46](https://user-images.githubusercontent.com/99462020/173605113-6a3b88d4-b043-4d46-8c40-6ac75d935d8d.png)

**Comment by abpei** [comment]
For testing I injected season/episode data into the guide for affected shows but the problem remained the same. The function for duplicate show removal just checks for the ShowId and removes all duplicates (which makes sense as the ShowId usually contains the show and episode number seperated by a dot). Now, some shows get published in guides without an episode identifier and they all simply come in as ".0000". That seems to be the case for many news and sports series. Consequently, that removal function cancels all duplicates which is what you see when all but the first occurrence get marked as cancelled (greyed out icon). When that recording completes and a schedule refresh runs in the meantime, you may get lucky enough that it schedules the next episode, but that doesn't work for shows that run multiple times per day. I would say it's debatable whether this is a JF bug or an issue with the guide provider. Ideally, we would inject the new-tag into the timer and skip those recordings from duplicate removal that are tagged as new, but you are going down a rabbit hole. For now, I submitted a pull request for what I think resolves this issue, by skipping shows from duplicate removal action when the show subkey is missing. It's been working fine for me for a while now. Let's see what the bosses think..

Screenshot from an example in the guide xml for clarification:
![image](https://user-images.githubusercontent.com/5302376/188920311-bb78fd15-9fce-443e-9c67-ed42b486f7a8.png)

**Comment by melendyk** [comment]
I disagree with it being debatable as to if it's a JF bug.  When I choose "Record ALL Episodes" and UNCHECK "Don't record episodes already in my library" it should record ALL episodes REGARDLESS if duplicate.  However, if I CHECK the box to not record episodes already in my library, then duplicates would be removed IF it's already IN my library.  Simple logic, but I digress.

**Comment by abpei** [comment]
When you unselect the "Don't record episodes already in my library" it skips the check for existence in the library and will schedule the recording even though it already exists in the library. Then the duplicate recording check hits and kills it anyway.. I'm not sure why the function is needed, but it's a leftover from Emby as far as I can tell.
